pkg/util/unstructured.go

```diff
@@ -182,3 +187,32 @@ func CheckAPIisPresent(dc *discovery.DiscoveryClient, gvr schema.GroupVersionRes
 
 	return false
 }
+
+// CreateStatusPatch creates a status patch for unstructured object.
+func CreateStatusPatch(unstrObj *unstructured.Unstructured, status map[string]interface{}) *unstructured.Unstructured {
+	patchedObj := &unstructured.Unstructured{}
+	patchedObj.SetAPIVersion(unstrObj.GetAPIVersion())
+	patchedObj.SetKind(unstrObj.GetKind())
+	patchedObj.SetName(unstrObj.GetName())
+	patchedObj.SetNamespace(unstrObj.GetNamespace())
+	patchedObj.Object["status"] = status
+	return patchedObj
+}
+
+// PatchStatus updates the object status with Patch.
+func PatchStatus(ctx context.Context, unstrObj *unstructured.Unstructured, status map[string]interface{},
+	namespace string, gvr schema.GroupVersionResource, dynamicClient dynamic.Interface) error {
+
+	patchBytes, err := json.Marshal(CreateStatusPatch(unstrObj, status))
+	if err != nil {
+		return err
+	}
+
+	if namespace != "" {
+		_, err = dynamicClient.Resource(gvr).Namespace(namespace).Patch(ctx, unstrObj.GetName(), types.MergePatchType, patchBytes, metav1.PatchOptions{})
+	} else {
+		_, err = dynamicClient.Resource(gvr).Patch(ctx, unstrObj.GetName(), types.MergePatchType, patchBytes, metav1.PatchOptions{})
+	}
```

Comment:
why not using in function `UpdateStatus`?

Reply:
When this func is called, it is already known that the generated UpdateStatus method failed.